FMS/affinity/affinity.c

```diff
@@ -27,6 +27,12 @@
 #include <sys/resource.h>
 #include <sys/syscall.h>
 
+static pid_t gettid1(void)
+{
+#ifndef __APPLE__
+  return syscall(SYS_gettid);
+#endif
+}
```

Comment:
I don't know the implications of changing this, so if you need this file reviewed you should ask someone on the team who does.

Reply:
The current build relies on system-defined gettid function that isn't present in the SLES operating system version installed on the Cray supercomputer.  A way around this is to add our own (or wrap a call to something that does work on the Cray -which is what we have here).  @ofuhrer made this fix originally in the build_daint branch.  I've successfully ran this fix on Piz Daint and through CircleCI (via a manual push of the testing).